vault/activity_log.go

```diff
@@ -0,0 +1,108 @@
+package vault
+
+import (
+	"context"
+	"os"
+	"sync"
+	"time"
+
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/sdk/logical"
+	"github.com/hashicorp/vault/vault/activity"
+)
+
+const (
+	// activitySubPath is the directory under the system view where
+	// the log will be stored.
+	activitySubPath = "activity/"
+)
+
+// ActivityLog tracks unique entity counts and non-entity token counts.
+// It handles assembling log fragments (and sending them to the active
+// node), writing log segments, and precomputing queries.
+type ActivityLog struct {
+	// log destination
+	logger log.Logger
+
+	// view is the storage location used by ActivityLog,
+	// defaults to sys/activity.
+	view logical.Storage
+
+	// nodeID is the ID to use for all fragments that
+	// are generated.
+	// TODO: use secondary ID when available?
+	nodeID string
+
+	// current log fragment (may be nil) and a mutex to protect it
+	fragmentLock     sync.RWMutex
+	fragment         *activity.LogFragment
+	fragmentCreation time.Time
+}
+
+// NewActivityLog creates an activity log.
+func NewActivityLog(_ context.Context, logger log.Logger, view logical.Storage) (*ActivityLog, error) {
```

Comment:
Curious, why have an unused parameter in a constructor? Can we just remove it or does it need to fulfill some interface?

Reply:
It's a placeholder, it will be needed as soon as we start reading from the storage on initialization.